driver/custom_endpoint_monitor.cc

```diff
@@ -0,0 +1,207 @@
+                                                 const std::string& endpoint_identifier, const std::string& region,
+                                                 long long refresh_rate_nanos, ctpl::thread_pool& thread_pool,
+                                                 bool enable_logging)
+    : topology_service(topology_service),
+      custom_endpoint_host(custom_endpoint_host),
+      endpoint_identifier(endpoint_identifier),
+      region(region),
+      refresh_rate_nanos(refresh_rate_nanos),
+      thread_pool(thread_pool),
+      enable_logging(enable_logging) {
+  if (enable_logging) {
+    this->logger = init_log_file();
+  }
+
+  this->run();
+}
+
+#ifdef UNIT_TEST_BUILD
+CUSTOM_ENDPOINT_MONITOR::CUSTOM_ENDPOINT_MONITOR(const std::shared_ptr<TOPOLOGY_SERVICE> topology_service,
+                                                 const std::string& custom_endpoint_host,
+                                                 const std::string& endpoint_identifier, const std::string& region,
+                                                 long long refresh_rate_nanos, ctpl::thread_pool& thread_pool,
+                                                 bool enable_logging, std::shared_ptr<Aws::RDS::RDSClient> client)
+    : topology_service(topology_service),
+      custom_endpoint_host(custom_endpoint_host),
+      endpoint_identifier(endpoint_identifier),
+      region(region),
+      refresh_rate_nanos(refresh_rate_nanos),
+      thread_pool(thread_pool),
+      enable_logging(enable_logging) {
+  if (enable_logging) {
+    this->logger = init_log_file();
+  }
+  this->run();
+}
+#endif
+
+bool CUSTOM_ENDPOINT_MONITOR::should_dispose() { return true; }
+
+bool CUSTOM_ENDPOINT_MONITOR::has_custom_endpoint_info() const {
+  auto default_val = std::shared_ptr<CUSTOM_ENDPOINT_INFO>(nullptr);
+  return custom_endpoint_cache.get(this->custom_endpoint_host, default_val) != default_val;
+}
+
+void CUSTOM_ENDPOINT_MONITOR::run() {
+  MYLOG_TRACE(this->logger, 0, "Starting custom endpoint monitor for '%s'", this->custom_endpoint_host.c_str());
+  if (thread_pool.size() == 1) {
+    // Each monitor should only have 1 thread.
+    return;
+  }
+  thread_pool.resize(1);
+  thread_pool.push([=](int id) {
+    ++SDK_HELPER;
+
+    Aws::RDS::RDSClientConfiguration client_config;
+    if (!region.empty()) {
+      client_config.region = region;
+    }
+
+    const Aws::RDS::RDSClient rds_client(Aws::Auth::DefaultAWSCredentialsProviderChain().GetAWSCredentials(),
+                                         client_config);
+    Aws::RDS::Model::Filter filter;
+    filter.SetName("db-cluster-endpoint-type");
+    filter.AddValues("custom");
+
+    Aws::RDS::Model::DescribeDBClusterEndpointsRequest request;
+    request.SetDBClusterEndpointIdentifier(this->endpoint_identifier);
+    // TODO: Investigate why filters returns `InvalidParameterCombination` error saying filter values are null.
+    // request.AddFilters(filter);
```

Comment:
Will uncomment this once this fix is released on aws-sdk-cpp: https://github.com/aws/aws-sdk-cpp/issues/3264